kernel/arch/dreamcast/include/dc/pvr/pvr_context.h

```diff
@@ -0,0 +1,612 @@
+    \ingroup                        pvr_ctx_attrib
+
+    These control how primitives are clipped against the user clipping area.
+*/
+typedef enum __packed pvr_clip {
+    PVR_USERCLIP_DISABLE = 0,   /**< \brief Disable clipping */
+    PVR_USERCLIP_INSIDE  = 2,   /**< \brief Enable clipping inside area */
+    PVR_USERCLIP_OUTSIDE = 3    /**< \brief Enable clipping outside area */
+} pvr_clip_t;
+
+/** \defgroup pvr_ctx_color     Color
+    \brief                      Color attributes for PowerVR primitive contexts
+    \ingroup                    pvr_ctx_attrib
+*/
+
+/** \brief     Texture U/V flipping mode for the PVR
+    \ingroup   pvr_ctx_texture
+
+    These flags determine what happens when U/V coordinate values exceed 1.0.
+    In any of the flipped cases, the specified coordinate value will flip around
+    after 1.0, essentially mirroring the image. So, if you displayed an image
+    with a U coordinate of 0.0 on the left hand side and 2.0 on the right hand
+    side with U flipping turned on, you'd have an image that was displayed twice
+    as if mirrored across the middle. This mirroring behavior happens at every
+    unit boundary (so at 2.0 it returns to normal, at 3.0 it flips, etc).
+
+    The default case is to disable mirroring. In addition, clamping of the U/V
+    coordinates by PVR_UVCLAMP_U, PVR_UVCLAMP_V, or PVR_UVCLAMP_UV will disable
+    the mirroring behavior.
+*/
+typedef enum __packed pvr_uv_flip {
+    PVR_UVFLIP_NONE = 0,  /**< \brief No flipped coordinates */
+    PVR_UVFLIP_V    = 1,  /**< \brief Flip V only */
+    PVR_UVFLIP_U    = 2,  /**< \brief Flip U only */
+    PVR_UVFLIP_UV   = 3   /**< \brief Flip U and V */
+} pvr_uv_flip_t;
+
+/** \brief   Enable or disable clamping of U/V on the PVR
+    \ingroup pvr_ctx_texture
+
+    These flags determine whether clamping will be applied to U/V coordinate
+    values that exceed 1.0. If enabled, these modes will explicitly override the
+    flip/mirroring modes (PVR_UVFLIP_U, PVR_UVFLIP_V, and PVR_UVFLIP_UV), and
+    will instead ensure that the coordinate(s) in question never exceed 1.0.
+*/
+typedef enum __packed pvr_uv_clamp {
+    PVR_UVCLAMP_NONE = 0,  /**< \brief Disable clamping */
+    PVR_UVCLAMP_V    = 1,  /**< \brief Clamp V only */
+    PVR_UVCLAMP_U    = 2,  /**< \brief Clamp U only */
+    PVR_UVCLAMP_UV   = 3   /**< \brief Clamp U and V */
+} pvr_uv_clamp_t;
+
+/** \brief    PowerVR texture sampling modes
+    \ingroup  pvr_ctx_texture
+*/
+typedef enum __packed pvr_filter {
+    PVR_FILTER_NONE       = 0,   /**< \brief No filtering (point sample) */
+    PVR_FILTER_NEAREST    = 0,   /**< \brief No filtering (point sample) */
+    PVR_FILTER_BILINEAR   = 2,   /**< \brief Bilinear interpolation */
+    PVR_FILTER_TRILINEAR1 = 4,   /**< \brief Trilinear interpolation pass 1 */
+    PVR_FILTER_TRILINEAR2 = 6    /**< \brief Trilinear interpolation pass 2 */
+} pvr_filter_t;
```

Comment:
I would actually use 0/1/2/3 here, because it is clear that bit 0 does not encode for the filter; and fixing it in pvr_poly_compile() and friends. That shouldn't break the API.